src/main/java/com/google/error_analyzer/backend/LogDao.java

```diff
@@ -0,0 +1,147 @@
+/**Copyright 2019 Google LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.*/
+
+package com.google.error_analyzer.backend;
+
+
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.*;
+import org.apache.http.HttpHost;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+import org.elasticsearch.action.admin.indices.get.GetIndexRequest;
+import org.elasticsearch.action.get.GetRequest;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.search.SearchResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.index.query.MatchQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.index.query.SimpleQueryStringBuilder;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.fetch.subphase.highlight.HighlightBuilder;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+
+
+public class LogDao implements DaoInterface {
+
+    private static final RestHighLevelClient client = new RestHighLevelClient
+        (RestClient.builder(new HttpHost("localhost", 9200, "http")));
+    private static final SearchSourceBuilder searchSourceBuilder 
+        = new SearchSourceBuilder();
+    private static final int windowSize = 10;
+    private static final Logger logger = LogManager.getLogger(LogDao.class);
+
+    //search db using keywords and return searchHits having highlight field added  
+    @Override 
+    public ImmutableList < SearchHit > fullTextSearch
+    (String fileName, String searchString, String field)throws IOException 
+    {
```

Comment:
You can keep the structure like this : 
public ImmutableList < SearchHit > fullTextSearch(
String fileName, String searchString, String field) throws IOException {